src/commands/dev/lib/dev-router.ts

```diff
@@ -33,29 +52,22 @@ export default function(reqPath = '', routes?: RouteConfig[]): RouteResult {
 
       if (match) {
         let destPath: string = reqPathname;
+        let headerLocation: string = "";
+
         if (routeConfig.dest) {
-          destPath = routeConfig.dest.replace(
-            /\$([1-9a-zA-Z]+)/g,
-            (_, param) => {
-              let matchIndex: number = keys.indexOf(param);
-              if (matchIndex === -1) {
-                // It's a number match, not a named capture
-                matchIndex = parseInt(param, 10);
-              } else {
-                // For named captures, add one to the `keys` index to
-                // match up with the RegExp group matches
-                matchIndex++;
-              }
-              return match[matchIndex];
-            }
-          );
+          destPath = resolveRouteParameters(routeConfig.dest, {match, keys})
+        }
+        
+        if (routeConfig.headers && routeConfig.headers.Location) {
+
+          headerLocation = resolveRouteParameters(routeConfig.headers.Location, {match, keys});
         }
 
         if (isURL(destPath)) {
           found = {
             dest: destPath,
             status: routeConfig.status,
-            headers: routeConfig.headers,
+            headers: (headerLocation === "") ? routeConfig.headers : Object.assign({},routeConfig.headers,{Location: headerLocation}),
```

Comment:
Instead of doing this logic in two places, can you move it up into a new variable?
I think it can be simplified as long as you don't mind modifying the original object.

```
let { headers } = routeConfig;
if (headers && headers.Location) {
  headers.Location = resolveRouteParameters(headers.Location, {match, keys});
}
```

Reply:
@styfle doing this will mutate routeConfig

Reply:
```js
let { headers } = routeConfig;
if (headers && headers.Location) {
  headers = { ...headers, Location: resolveRouteParameters(headers.Location, {match, keys}) };
}
```